services/src/main/java/org/keycloak/services/resources/admin/ClientResource.java

```diff
@@ -221,8 +222,12 @@ public void deleteClient() {
             throw new ErrorResponseException(cpe.getError(), cpe.getErrorDetail(), Response.Status.BAD_REQUEST);
         }
 
-        new ClientManager(new RealmManager(session)).removeClient(realm, client);
-        adminEvent.operation(OperationType.DELETE).resourcePath(session.getContext().getUri()).success();
+        if (new ClientManager(new RealmManager(session)).removeClient(realm, client)) {
+            adminEvent.operation(OperationType.DELETE).resourcePath(session.getContext().getUri()).success();
+        }
+        else {
+            throw new ForbiddenException();
```

Comment:
I think we shouldn't throw the `ForbiddenException` here. Please take a look at `ErrorResponseException` instead.

Reply:
Could you maybe explain why it would be better to throw an ErrorResponseException here ? A ForbiddenException seems fitting for me in this particular case.

Reply:
I changed it to be a ErrorResponseException. I'm assuming the reason for it is that you can't be entirely sure that the reason for failure is an internal client. Therefore i included a "Bad Request" response into the ErrorResponseException. This seems to be consistent with the other ones.

Reply:
That's right, also in UI you get nice error message instead of "Unexpected error"